File: core/internal/dtos/access_dto.go
package dtos

import "errors"

type AccessDataRequest struct {
	AppID       string `json:"appId"`
	Secret      string `json:"secret"`
	Scope       string `json:"scope"`
	VaultDataID string `json:"vaultDataId"`
}

func (r *AccessDataRequest) Validate() error {
	if r.AppID == "" {
		return errors.New("appId is required")
	}
	if r.Secret == "" {
		return errors.New("secret is required")
	}
	if r.Scope == "" {
		return errors.New("scope is required")
	}
	if r.VaultDataID == "" {
		return errors.New("vaultDataId is required")
	}
	return nil
}

File: gateway/src/dtos/access.dto.js
class AccessDto {
    static validate(req) {
        const { appId, secret, scope, vaultDataId } = req.body;

        if (!req.isOpaqueToken) {
            if (!appId || typeof appId !== 'string') {
                throw new Error('appId is required and must be a string');
            }
            if (!secret || typeof secret !== 'string') {
                throw new Error('secret is required and must be a string');
            }
        }
        if (!scope || typeof scope !== 'string') {
            throw new Error('scope is required and must be a string');
        }
        if (!vaultDataId || typeof vaultDataId !== 'string') {
            throw new Error('vaultDataId is required and must be a string');
        }
    }
}

module.exports = AccessDto;
